doc/cephfs/app-best-practices.rst

```diff
@@ -0,0 +1,83 @@
+
+Application best practices for distributed filesystems
+======================================================
+
+CephFS is POSIX compatible, and therefore should work with any existing
+applications that expect a POSIX filesystem.  However, because it is a
+network filesystem (unlike e.g. XFS) and it is highly consistent (unlike
+e.g. NFS), there are some consequences that application authors may
+benefit from knowing about.
+
+The following sections describe some areas where distributed filesystems
+may have noticeably different performance behaviours compared with
+local filesystems.
+
+
+ls -l (and colored ls)
+----------------------
+
+When you run "ls -l" (or use ls with coloring enabled), the ``ls`` program
+is first doing a directory listing, and then calling ``stat`` on every
+file in the directory.
+
+This is usually far in excess of what an application really needs, and
+it can be slow for large directories.  If you don't really need all
+this metadata for each file, then use a plain ``ls`` with coloring
+disabled.
```

Comment:
I'm not sure this the case for ls with coloring. `ls` gets enough information from `getdents` to figure out the types of files (`d_type`) for coloring. In my own test just now, it only does `getdents`.